kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/PartitionUtils.java

```diff
@@ -133,6 +133,50 @@ public static Tuple2<Predicate, Predicate> splitMetadataAndDataPredicates(
         }
     }
 
+    /**
+     * Rewrite the given predicate on partition columns on `partitionValues_parsed` in checkpoint
+     * schema. The rewritten predicate can be pushed to the Parquet reader when reading the
+     * checkpoint files.
+     *
+     * @param predicate Predicate on partition columns.
+     * @param partitionColMetadata Map of partition column name (in lower case) to its type.
+     * @return Rewritten {@link Predicate} on `partitionValues_parsed` in `add`.
+     */
+    public static Predicate rewritePartitionPredicateOnCheckpointFileSchema(
+            Predicate predicate,
+            Map<String, StructField> partitionColMetadata) {
```

Comment:
nit: I think `metadata` is an overloaded term. Could we just call this `partitionColToField` ?